src/ldapclient_lib.py

```diff
@@ -0,0 +1,395 @@
+        ldbd_secret = self.model.get_secret(label="ldap-default-bind-dn")
+        lp_secret = self.model.get_secret(label="ldap-password")
+        ca_cert_secret.remove_all_revisions()
+        ldbd_secret.remove_all_revisions()
+        lp_secret.remove_all_revisions()
+        self.on.server_unavailable.emit()
+
+    def _on_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Event emitted when the relation is joined.
+
+        Looks at the relation data and config values and emits:
+        - config unavailable event: If the config resource is not supplied.
+        - glauth snap event: When the glauth certificate, config, and key are available.
+        """
+        self.charm.unit.status = MaintenanceStatus("reconfiguring glauth")
+        # Check model for GLAuth config resource
+        try:
+            resource_path = self.model.resources.fetch("config")
+        except ModelError:
+            logger.debug("No config resource supplied")
+            resource_path = None
+            self.on.config_data_unavailable.emit(
+                api_port=self.model.config["api-port"], ldap_port=self.model.config["ldap-port"]
+            )
+        # Set config and get LDAP URI
+        ldap_uri = self.set_config(
+            resource_path, self.model.config["ldap-port"], self.model.config["api-port"]
+        )
+        # Get CA Cert and key
+        ca_cert = self.load()
+        # Signals glauth snap is ready to be started
+        self.on.glauth_snap_ready.emit()
+        cc_content = {"ca-cert": ca_cert}
+        # Get Peer Secrets
+        ldap_relation = self.model.get_relation("glauth")
+        default_bind_dn = ldap_relation.data[self.charm.app]["ldap-default-bind-dn"]
+        ldap_password = ldap_relation.data[self.charm.app]["ldap-password"]
+        ldbd_secret = self.model.get_secret(id=default_bind_dn)
+        lp_secret = self.model.get_secret(id=ldap_password)
+        # Create Secrets
+        cc_secret = self.charm.app.add_secret(cc_content, label="ca-cert")
+        logger.debug("created secret %s", cc_secret)
+        cc_secret.grant(event.relation)
+        ldbd_secret.grant(event.relation)
+        lp_secret.grant(event.relation)
+        event.relation.data[self.charm.app]["ca-cert"] = cc_secret.id
+        event.relation.data[self.charm.app]["ldap-default-bind-dn"] = ldbd_secret.id
+        event.relation.data[self.charm.app]["ldap-password"] = lp_secret.id
+        # Configuration data update
+        ldap_relation = self.model.get_relation("ldap-client")
+        ldap_relation.data[self.charm.app].update(
+            {
+                "basedn": self.model.config["ldap-search-base"],
+                "ldap-uri": ldap_uri,
+            }
+        )
+        self.charm.unit.status = ActiveStatus("glauth ready")
+
+    def load(self) -> str:
+        """Load ca-certificate from glauth snap.
+
+        Returns:
+            The ca certificate content.
```

Comment:
Should specify the return type:

```suggestion
            str: The ca certificate content.
```